index.bs

```diff
@@ -242,6 +244,77 @@ The <dfn attribute for=FileSystemHandle>name</dfn> getter steps are to return [=
   :: Returns true if |handle1| and |handle2| represent the same file or directory.
 </div>
 
+### The {{FileSystemHandle/remove()}} method ### {#api-filesystemhandle-remove}
+
+<div class="note domintro">
+  : await |handle| . {{FileSystemHandle/remove()|remove}}()
+  : await |handle| . {{FileSystemHandle/remove()|remove}}({ {{FileSystemRemoveOptions/recursive}}: false })
+  :: Attempts to remove the entry represented by |handle| from the underlying file system.
+
+     Attempting to delete a non-empty directory will result in a promise rejection.
+
+  : await |handle| . {{FileSystemHandle/remove()|remove}}({ {{FileSystemRemoveOptions/recursive}}: true })
+  :: Attempts to remove the entry represented by |handle| from the underlying file system.
+
+     If the entry is a directory, its contents will also be deleted recursively.
+
+     For files or directories with multiple hardlinks or symlinks, the entry which is deleted
+     is the entry corresponding to the path which was used to obtain the handle.
+</div>
+
+<div algorithm>
+The <dfn method for=FileSystemHandle>remove(|options|)</dfn> method steps are:
+
+1. Let |result| be [=a new promise=].
+1. Let |entry| be [=this=]'s [=FileSystemHandle/entry=].
+1. Run these steps [=in parallel=]:
+  1. Let |access| be the result of running |entry|'s
+     [=file system entry/request access=] given "`readwrite`".
+  1. If |access| is not "{{PermissionState/granted}}":
+    1. [=Queue a storage task=] given |entry|'s [=relevant global object=] to
+        [=/reject=] |result| with a "{{NotAllowedError}}" {{DOMException}}.
+    1. Abort these steps.
```

Comment:
These should probably also be part of the lockResultAlgorithm? No need for nested in parallel I'd think.

Reply:
Hmm. The "request access" check should happen before attempting to take the lock, and we shouldn't bother attempting to take the lock unless the "request access" check succeeds. So I'm not sure how this would fit into the `lockResultAlgorithm`?

Should we also post the "request access" algorithm as a task? Something like:
```
(in parallel)
1. Let |lockResultAlgorithm| be an algorithm which takes a |lockResult| and
     runs these steps:
  <do file system operations>
  ...

1. Let |accessResultAlgorithm| be an algorithm which takes an |accessResult| and
     runs these steps:
  1. Reject |result| if not "granted"
  1. Take a lock on |entry| with |lockResultAlgorithm|.

1. Queue a storage task given |entry|'s relevant global object to run |entry|'s request access algorithm given "readwrite"
```

Reply:
I guess you're saying that we want other lock requests to be able to happen while we do request access checks? In that case this is probably what you want.

Though what might be sufficient as I don't think this is really observable is that instead of a "file system lock queue" we have a "file system queue" or some such and all file system operations go through there.

Reply:
(By the way, I don't always immediately check PRs as I expect to have to review. If you just want to hash something out https://whatwg.org/chat will likely be quicker.)

Reply:
Updated #95 to post all file system operations to a "file system queue", as we discussed offline. Once that LGTY I'll update this PR, as well. Thanks for the suggestion!

Reply:
#95 has since merged 🎉

I updated this PR to follow the new paradigm